cvmfs/catalog_mgr_rw.cc

```diff
@@ -56,7 +56,12 @@ WritableCatalogManager::WritableCatalogManager(
 {
   sync_lock_ =
     reinterpret_cast<pthread_mutex_t *>(smalloc(sizeof(pthread_mutex_t)));
-  int retval = pthread_mutex_init(sync_lock_, NULL);
+  pthread_mutexattr_t attr;
```

Comment:
The recursive lock is there to allow `WritableCatalogManager::SwapNestedCatalog` to call `WritableCatalogManager::RemoveNestedCatalog`, to avoid code duplication.  It would not be required if the code is restructured to avoid calling `WritableCatalogManager` methods.